### PR TITLE
Fixes an issue where bazel could not create runfiles on windows.

### PR DESCRIPTION
**Pull request by lukasholzer**
This issue is due to some legacy run files linking where a manifest file was expected where it was not on windows.
It constantly failed with:

```
ERROR: Manifest file c:\Users\ADMINI~1\_BAZEL~1\rgyky3dz\execroot\EXAMPL~1\bazel-out\x64_windows-fastbuild\bin\test.bat.runfiles\examples_jest\test.bat.runfiles_manifest does not exist.
```

Issue described here: https://github.com/bazelbuild/rules_nodejs/issues/1454

The only thing that might be needed under windows is a `.bazelrc.user` where the output path is set as there might be some issues with the max path length on windows.
https://docs.bazel.build/versions/3.5.0/windows.html#avoid-long-path-issues

```
startup --output_user_root=C:/tmp
```

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by lukasholzer** [comment]
@SkaillZ fixed it for the rules_nodejs as well ;) https://github.com/bazelbuild/rules_nodejs/pull/2178

**Comment by lukasholzer** [comment]
patch file has to be in there until https://github.com/stylelint/stylelint/issues/4929 and https://github.com/sindresorhus/globby/issues/152 are resolved for stylelint on windows.

**Comment by lukasholzer** [comment]
image

testing with jest and stylelint worked out great now on windows.

**Comment by christian-fischer** [comment]
just tried `bazel run //libs/barista-components/button:test` and `bazel run //libs/barista-components/button:stylelint` and it works fine on my windows machine.

Also any errors are now nicely visible on the log:
```
Error: Lint errors found in the listed files:
ÔØî libs\barista-components\button\src\_button-theme.scss ,
Ô£à libs\barista-components\button\src\button.scss
L6: Delete "ÔÅÄ" (prettier/prettier)
L5: Expected class selector ".foo" to match specified pattern (selector-class-pattern)
    at C:\tools\bazel_rules\stylelint\run-stylelint.ts:64:11
    at Generator.next (<anonymous>)
    at fulfilled (C:\tmp\46maqhrj\external\npm\node_modules\tslib\tslib.js:111:62)
```

File: patches/jest-haste-map+26.1.0.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
+diff --git a/node_modules/jest-haste-map/build/crawlers/node.js b/node_modules/jest-haste-map/build/crawlers/node.js
+index 6b6e11a..2ff08fa 100644
+--- a/node_modules/jest-haste-map/build/crawlers/node.js
++++ b/node_modules/jest-haste-map/build/crawlers/node.js
+@@ -205,7 +205,11 @@ function find(roots, extensions, ignore, callback) {
+ 
+ function findNative(roots, extensions, ignore, callback) {
+   const args = Array.from(roots);
```
Comment (heartdisease) [inline patches/jest-haste-map+26.1.0.patch:8]: Wouldn't this be more readable?

`const args = [...Array.from(roots), '(', '-type', 'f', '-o', '-type', 'l', ')'];`

**Review by heartdisease — APPROVED** [review]

**Review by heartdisease — APPROVED** [review]

**Pull request merged**